File: sample/Python/kabusapi_sendorder_option_pay_ClosePositionOrder.py
import urllib.request
import json
import pprint

obj = { 'Password': '123456',
        'Symbol': '145123218',
        'Exchange': 23,
        'TradeType': 2,
        'TimeInForce': 2,
        'Side': '2',
        'Qty': 1,
        'ClosePositionOrder': 1,
        'FrontOrderType': 30,
        'Price': 0,
        'ExpireDay': 0,
        'ReverseLimitOrder': {
                               'TriggerPrice': 1150,
                               'UnderOver': 1, #1.以下 2.以上
                               'AfterHitOrderType': 1, #1.成行 2.指値
                               'AfterHitPrice': 0
                             }
      }
json_data = json.dumps(obj).encode('utf-8')

url = 'http://localhost:18080/kabusapi/sendorder/option'
req = urllib.request.Request(url, json_data, method='POST')
req.add_header('Content-Type', 'application/json')
req.add_header('X-API-KEY', 'ed94b0d34f9441c3931621e55230e402')

try:
    with urllib.request.urlopen(req) as res:
        print(res.status, res.reason)
        for header in res.getheaders():
            print(header)
        print()
        content = json.loads(res.read())
        pprint.pprint(content)
except urllib.error.HTTPError as e:
    print(e)
    content = json.loads(e.read())
    pprint.pprint(content)
except Exception as e:
    print(e)

File: sample/C#/CSharp/Kabusapi_Sendorder_Option_pay_ClosePositionOrder.cs
using Newtonsoft.Json;
using System;
using System.Net.Http;
using System.Text;

namespace CSharp_sample
{
    class Kabusapi_Sendorder_Option_pay_ClosePositionOrder
    {
        static void Main(string[] args)
        {
            var obj = new
            {
                Password = "111111",
                Symbol = "145123218",
                Exchange = 23,
                TradeType = 2,
                TimeInForce = 2,
                Side = "2",
                Qty = 1,
                ClosePositionOrder = 1,
                FrontOrderType = 30,
                Price = 0,
                ExpireDay = 20200925,
                ReverseLimitOrder = new
                {
                    TriggerPrice = 1150,
                    UnderOver = 1,
                    AfterHitOrderType = 1,
                    AfterHitPrice = 0
                }

            };
            var url = "http://localhost:18080/kabusapi/sendorder/option";
            string token = GenerateToken.GetToken();

            try
            {
                var client = new HttpClient();
                var request = new HttpRequestMessage(HttpMethod.Post, url);
                request.Headers.Add("ContentType", "application/json");
                request.Headers.Add("X-API-KEY", token);
                request.Content = new StringContent(JsonConvert.SerializeObject(obj), Encoding.UTF8, "application/json");

                HttpResponseMessage response = client.SendAsync(request).Result;
                Console.WriteLine("{0} \n {1}", JsonConvert.DeserializeObject(response.Content.ReadAsStringAsync().Result), response.Headers);

            }
            catch (HttpRequestException e)
            {
                Console.WriteLine("{0} {1}", e, e.Message);
                Console.ReadKey();
            }
            catch (Exception ex)
            {
                Console.WriteLine("{0} {1}", ex, ex.Message);
                Console.ReadKey();
            }

            Console.ReadKey();
        }
    }
}
